agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/impl/AgentRegistryImpl.java

```diff
@@ -57,23 +50,16 @@
  * @author Anatoliy Bazko
  */
 @Singleton
-public class LocalAgentRegistryImpl implements AgentRegistry {
+public class AgentRegistryImpl implements AgentRegistry {
 
-    protected static final Logger  LOG    = LoggerFactory.getLogger(LocalAgentRegistryImpl.class);
-    private static final   Pattern AGENTS = Pattern.compile(".*[//]?agents/[^//]+[.]json");
+    protected static final Logger LOG = LoggerFactory.getLogger(AgentRegistryImpl.class);
 
     private final Map<String, Agent> agents;
-    private final List<String>       agentIds;
 
     @Inject
-    public LocalAgentRegistryImpl(Set<Agent> agents) throws IOException {
-
-        this.agents = new HashMap<>();
-        findAgents();
-        for (Agent agent : agents) {
-            this.agents.put(agent.getId(), agent);
-        }
-        this.agentIds = ImmutableList.copyOf(this.agents.keySet());
+    public AgentRegistryImpl(Set<Agent> agents) throws IOException {
+        this.agents = new HashMap<>(agents.size());
+        agents.forEach(a -> this.agents.put(a.getId(), a));
```

Comment:
Here it is possible that someone declared 2 equal agents. 
If all of them have equals and hashcode then multibinder will throw an error.
If multibinder is declared anywhere with support of duplicates then it won't throw exception.
If equals and hashcode is not overridden or overridden in a different ways then here duplicate agent will override previous one.